pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -212,11 +212,11 @@ func (c *Controller) GetService(hostname host.Name) (*model.Service, error) {
 // any of the supplied labels. All instances match an empty label list.
 func (c *Controller) InstancesByPort(svc *model.Service, port int,
 	labels labels.Collection) ([]*model.ServiceInstance, error) {
-	var instances, tmpInstances []*model.ServiceInstance
+	instances := make([]*model.ServiceInstance, 0, len(c.GetRegistries()))
```

Comment:
this one is also not quite right

Reply:
Done. Yup it appends multiples slices to length is dynamic